src/Microsoft.HttpRepl.Tests/HttpStateTests.cs

```diff
@@ -49,6 +53,132 @@ public void GetPathString_MultiplePathSections_CorrectString()
             Assert.Equal(expected, result);
         }
 
+        [Fact]
+        public void GetApplicableContentTypes_NoStructure_ReturnsNull()
+        {
+            HttpState httpState = SetupHttpState();
+
+            httpState.BaseAddress = new Uri("https://localhost/");
+            httpState.Structure = null;
+
+            IEnumerable<string> result = httpState.GetApplicableContentTypes(null, string.Empty);
+
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void GetApplicableContentTypes_NoMethod_ReturnsAll()
+        {
+            HttpState httpState = SetupHttpState();
+
+            DirectoryStructure directoryStructure = new DirectoryStructure(null);
+            RequestInfo requestInfo = new RequestInfo();
+            requestInfo.SetRequestBody("GET", "application/json", "");
+            requestInfo.SetRequestBody("PUT", "application/xml", "");
+            directoryStructure.RequestInfo = requestInfo;
+
+            httpState.BaseAddress = new Uri("https://localhost/");
+            httpState.Structure = directoryStructure;
+
+            IEnumerable<string> result = httpState.GetApplicableContentTypes(null, "");
+
+            Assert.NotNull(result);
+
+            List<string> resultList = result.ToList();
+            Assert.Equal(2, resultList.Count);
+            Assert.Contains("application/json", resultList, StringComparer.OrdinalIgnoreCase);
+            Assert.Contains("application/xml", resultList, StringComparer.OrdinalIgnoreCase);
+        }
+
+        [Fact]
+        public void GetApplicableContentTypes_GetMethod_ReturnsCorrectOne()
+        {
+            HttpState httpState = SetupHttpState();
+
+            DirectoryStructure directoryStructure = new DirectoryStructure(null);
+            RequestInfo requestInfo = new RequestInfo();
+            requestInfo.SetRequestBody("GET", "application/json", "");
+            requestInfo.SetRequestBody("PUT", "application/xml", "");
+            directoryStructure.RequestInfo = requestInfo;
+
+            httpState.BaseAddress = new Uri("https://localhost/");
+            httpState.Structure = directoryStructure;
+
+            IEnumerable<string> result = httpState.GetApplicableContentTypes("GET", "");
+
+            Assert.NotNull(result);
+
+            List<string> resultList = result.ToList();
```

Comment:
Do we need to convert to list? Can we use Assert.Contains(..) instead?

Reply:
No, didn't need it. Was just there from copying the first couple tests where it was more useful. Removed in 624ea2c.